pypose/module/pnp.py

```diff
@@ -0,0 +1,493 @@
+import torch
+import pypose
+from pypose.optim import GaussNewton
+
+
+class EPnP(torch.nn.Module):
+    r"""
+        EPnP Solver - a non-iterative O(n) solution to the PnP problem.
+        as described in:
+
+        Francesc Moreno-Noguer, Vincent Lepetit, Pascal Fua.
+        Accurate Non-Iterative O(n) Solution to the PnP Problem.
+        In Proceedings of ICCV, 2007.
+        source: https://github.com/cvlab-epfl/EPnP
```

Comment:
move citation to end of doc, Use `Note:` when necessary. refer to [LQR](https://pypose.org/docs/main/generated/pypose.module.LQR)